app/frontend/actions.ts

```diff
@@ -982,7 +1030,132 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
+    try {
+      loadingAction(state, 'Loading account')
+      const nextAccount = await wallet.exploreNextAccount()
+      const accountInfo = await nextAccount.getAccountInfo(state.validStakepools)
+      const accountsInfo = [...state.accountsInfo, accountInfo]
+      setState({
+        accountsInfo,
+        ...getWalletInfo(accountsInfo),
+      })
+      setActiveAccount(state, nextAccount.accountIndex)
+    } catch (e) {
+      setErrorState('walletLoadingError', e)
+      setState({
+        shouldShowWalletLoadingErrorModal: true,
+      })
+    } finally {
+      stopLoadingAction(state, {})
+    }
+  }
+
+  const setTargetAccount = async (state: State, accountIndex: number) => {
+    setState({
+      targetAccountIndex: accountIndex,
+    })
+    const targetAddress = await wallet.getAccount(accountIndex).getChangeAddress()
+    updateAddress(state, null, targetAddress)
+  }
+
+  const setSourceAccount = async (state: State, accountIndex: number) => {
+    resetTransactionSummary(state)
+    setState({
+      sourceAccountIndex: accountIndex,
+    })
+    const targetAddress = await wallet.getAccount(getState().targetAccountIndex).getChangeAddress()
+    updateAddress(state, null, targetAddress)
+  }
+
+  const showSendTransactionModal = async (
+    state: State,
+    sourceAccountIndex: number,
+    targetAccountIndex: number
+  ) => {
+    resetTransactionSummary(state)
+    setState({
+      sourceAccountIndex,
+      targetAccountIndex,
+      sendTransactionTitle: 'Transfer funds between accounts',
+      shouldShowSendTransactionModal: true,
+      txSuccessTab: '',
+      sendAmount: {fieldValue: '', coins: 0},
+      transactionFee: 0,
+    })
+    const targetAddress = await wallet.getAccount(targetAccountIndex).getChangeAddress()
+    updateAddress(getState(), null, targetAddress)
+  }
+
+  const closeSendTransactionModal = (state: State) => {
+    resetTransactionSummary(state)
+    resetAccountIndexes(state)
+    setState({
+      sendAddress: {fieldValue: ''},
+      sendAmount: {fieldValue: '', coins: 0},
+      transactionFee: 0,
+      shouldShowSendTransactionModal: false,
+      sendAddressValidationError: null,
+      sendAmountValidationError: null,
+    })
+  }
+
+  const switchSourceAndTargetAccounts = async (state: State) => {
+    const targetAccountIndex = state.sourceAccountIndex
+    const sourceAccountIndex = state.targetAccountIndex
+    resetTransactionSummary(state)
+    setState({
+      sourceAccountIndex,
+      targetAccountIndex,
+    })
+    const targetAddress = await wallet.getAccount(targetAccountIndex).getChangeAddress()
+    updateAddress(state, null, targetAddress)
+  }
+
+  const showDelegationModal = (state: State, sourceAccountIndex: number) => {
+    setState({
+      sourceAccountIndex,
+      // TODO: move this title logic to the actual component
+      delegationTitle: state.shouldNumberAccountsFromOne
+        ? `Delegate Account #${sourceAccountIndex + 1} Stake`
```

Comment:
deciding whether to print `#${sourceAccountIndex + 1}` or just `sourceAccountIndex` imho deserves a helper function "formatAccountNumber" to centralize the logic of rendering it